### DISCUSSION TITLE
How to work with user relationships?

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#4374] [Q&A]
Hello, thanks for taking the time to answer my question :)

I'm making a self-hosted application in Sveltekit with the data shared by two (or more) users.

I made a field called `Spending` has a field called `RegisteredSpending` that stores which user registered such record.

I'm using the following TS code to get my records:
(This is testing code, pardon any heresies you read here)

```typescript
    try {
        const records = await locals.pb.collection('Spending').getFullList({
            sort: "-created",
            expand: "RegisteredSpending"
        });

        console.log(records)

        return {
            records: records
        }
    } catch (err) {
        console.log(err)
        throw err;
    }
```

However the only thing that produces a result for me is 

```json
  {
    Title: 'Test1',
    SpendingDate: '2024-02-16 12:00:00.000Z',
    AmountInCents: 12000,
    Note: 'This is a Test Record',
    PaymentForm: 'Something',
    RegisteredSpending: '15jr0l2c468ibnk',
    collectionId: 'randomid_xd',
    collectionName: 'Spending',
    created: '2024-02-17 18:25:52.465Z',
    expand: { RegisteredSpending: [Object] },
    id: '0kp16vrge3rm4zy',
    updated: '2024-02-17 18:25:52.465Z'
  }
```

I don't know what am I doing wrong here I tried following the docs and even making guesses but so far none have worked.

I already tried with:
- `expand: "RegisteredSpending(Spending)"`
- `expand: "RegisteredSpending(user).username"`
- `expand: "RegisteredSpending(username)"`
- `RegisteredSpending,user.username`
 
As alternatives but they keep sending the same response which is just the RegisteredSpending person ID.

Is there a straightforward way to get these details using relationships? Thanks for your answer :)

**Comment by ganigeorgiev** [comment]
The expand result will be under the `expand` object (it is included in your example).

**Reply by ghost** [reply]
Only the first returned object is getting expanded. In the docs I found this:

> Currently only single relation fields can be back-relation expanded (aka. when "Max Select" field option is 1).

Should I add that as extra parameter to my `getFullList({})` call?

Thank you so much for your response :bow:

**Reply by ganigeorgiev** [reply]
The cited note is for back-relations. In your use case, at least based on the example json, you have a direct relation from Spending->RegisteredSpending.

If the other items of the `records` array doesn't have an `expand` object then this means that your client request doesn't satisfy the View API rule of the related RegisteredSpending collection for those records.

**Reply by ghost** [reply]
Thank you! I guess I was not taking those permissions into consideration :facepalm: I'll try to guess a new approach to that. Thank you for your response!

**Comment by ghost** [comment]
Unmarking as solved. Discovered this while doing a similar approach with full permissions:

## Problem: 
Having a `Products` collection with the following structure:

```json
{
      "id": "RECORD_ID",
      "collectionId": "xxxxxxxxxx",
      "collectionName": "Products",
      "created": "2023-01-01 01:00:00.123Z",
      "updated": "2023-01-01 23:59:59.456Z",
      "Name": "test",
      "Price": 123,
      "Category": "RELATION_RECORD_ID" 
}
```
Where `Category` is a relation to `ProductsCategory`.

`ProductsCategory` being another collection:

```json
      "id": "RECORD_ID",
      "collectionId": "4c9topcis5o4le9",
      "collectionName": "ProductsCategory",
      "created": "2023-01-01 01:00:00.123Z",
      "updated": "2023-01-01 23:59:59.456Z",
      "Name": "test",
``` 

Doing any of these, returns no expansion, nor field at all. Permissions are open to everyone: 

```typescript
        const records = await locals.pb.collection('Products').getFullList({
            sort: "-created",
            expand: "Category"
        });
```

```typescript
        const records = await locals.pb.collection('Products').getFullList({
            sort: "-created",
            expand: "Category(Name)"
        });
```

```typescript
        const records = await locals.pb.collection('Products').getFullList({
            sort: "-created",
            expand: "Category(ProductsCategory)"
        });
```

My apologies for my previous `Solved` mark.

Use a string as a category and match the two table values manually. It's more code but you won't go bald earlier in life.
